kore-rpc-types/src/Kore/JsonRpc/Types.hs

```diff
@@ -259,6 +270,14 @@ instance Pretty.Pretty (API 'Req) where
         GetModel _ -> "get-model"
         Cancel -> "cancel"
 
+instance Pretty.Pretty (API 'Res) where
+    pretty = \case
+        Execute (ExecuteResult{reason, depth}) -> "execute" Pretty.<+> Pretty.pretty reason Pretty.<+> Pretty.pretty depth
+        Implies _ -> "implies"
+        Simplify _ -> "simplify"
+        AddModule _ -> "add-module"
+        GetModel _ -> "get-model"
```

Comment:
Certainly enough for the dry-run purpose ... but the printed information only provides the basics. 
Consider moving this to `kore-rpc-client`. (The `HaltReason` instance can, and should, stay here).
For a proper `Pretty` instance, I would expect at least the main results to be printed (like `Implies{valid}`, `{GetModel{satisfiable}`), if not all the terms in the response.